include/net/net_ip.h

```diff
@@ -208,6 +213,12 @@ struct sockaddr_ll_ptr {
 	u8_t        *sll_addr;    /* Physical-layer address  */
 };
 
+/** Socket address for UNIX domain sockets, pipes, etc */
+struct sockaddr_un {
+	sa_family_t sun_family;    /* AF_UNIX  */
+	char        sun_path[108]; /* Pathname */
```

Comment:
I think we need to make this configurable via Kconfig option. The value 108 is the same as in Linux but for Zephyr it is consuming way too much memory. For socketpair use, the path is not used according to manual `Likewise, the two sockets created by socketpair(2) are unnamed.  When the address of an unnamed socket is returned, its length is sizeof(sa_family_t), and sun_path should not be inspected.`
We could make the default value for path length so that sockaddr_un and sockaddr_in6 would be the same size. This way there is some space for path and user can increase the value if needed.

Reply:
@jukkar - yeah, I agree, it is a bit much, and technically speaking, it is unnecessary for socketpair(2) or pipe(2). Would it be better to remove this definition, since named sockets are not even part of the proposal? Originally, I was going to use the sun_path as either an actual path or a string of random bytes to assist in matching fds, but both of those things are probably unnecessary.